src/qt/walletmodel.cpp

```diff
@@ -563,3 +563,8 @@ bool WalletModel::isMultiwallet()
 {
     return m_node.getWallets().size() > 1;
 }
+
+void WalletModel::refresh(bool pk_hash_only)
+{
+    addressTableModel = new AddressTableModel(this, pk_hash_only);
```

Comment:
I'm not sure it makes sense to use the same cache with context-dependent factors like this.

Probably would be best to keep the same AddressTableModel with all addresses, and then a FilteredAddressTableModel(AddressTableModel*) to hide them...